### PR TITLE
[WIP] Endpoints and requests to send payment through Circle

### PR DESCRIPTION
**Pull request by jw122**
1. Load the API key as an env variable so it can be used in the `CircleClient`
2. Run the server
3. Try a `POST` request like this:
```
curl --location --request POST 'localhost:8080/payment' \
--header 'Content-Type: application/json' \
--data-raw '{"amount": "23.00", "description": "payment", "email": "hello@test.com", "phoneNumber": "+12025550180", "verificationMethod": "cvv", "cvv": "123", "sourceId": "bede6cc9-80b0-41b7-8d6f-970f5d3806c2", "sourceType": "card"}'
```

image

## TODO
- [x] put api key in .env file
- [x] cleanup + organization
- [ ] some fields into the `Payment` object should be optional
- [ ] CVV encryption

### CODE REVIEW

File: server/src/main/kotlin/circle/CircleClient.kt [hunk 1]
```diff
@@ -1,14 +1,74 @@
...
+}
+
+suspend fun makePayment(paymentRequest: PaymentRequest): String {
+    val client = HttpClient(CIO) {
+        install(JsonFeature) {
+            serializer =  KotlinxSerializer()
+        }
+    }
+    val response: HttpResponse = client.post("https://api-sandbox.circle.com/v1/payments") {
+        val apiKey : String = System.getenv("CIRCLE_API_KEY") ?: ""
```
Comment (jw122) [inline server/src/main/kotlin/circle/CircleClient.kt:47]: this is where the API key gets referenced

File: server/src/main/kotlin/circle/CircleClient.kt [hunk 2]
```diff
@@ -1,14 +1,76 @@
...
+
+suspend fun makePayment(paymentRequest: PaymentRequest): String {
+    val client = HttpClient(CIO) {
+        install(JsonFeature) {
+            serializer =  KotlinxSerializer()
+        }
+    }
+    val response: HttpResponse = client.post("https://api-sandbox.circle.com/v1/payments") {
+        val dotenv = dotenv()
+        val apiKey : String = dotenv["CIRCLE_API_KEY"]
```
Comment (jw122) [inline server/src/main/kotlin/circle/CircleClient.kt:49]: this is where the API key gets referenced

File: server/src/main/kotlin/routes/PaymentRoutes.kt [hunk 3]
```diff
@@ -36,25 +36,36 @@ fun Route.paymentRouting() {
...
+
         post {
             val payment = call.receive<Payment>()
+            println("payment request body: " + payment)
+            // Prepare a payment request
+            // TODO: extract IP address from request, user session ID, encrypt the CVV
+            val request = circle.buildPaymentRequest(
+                payment.sourceId,
+                payment.sourceType,
+                "172.33.222.1",
```
Comment (jw122) [inline server/src/main/kotlin/routes/PaymentRoutes.kt:55]: Trying to figure out how to get the IP address of an incoming request with ktor

**Comment by popoaq** [comment]
LGTM

File: server/src/main/kotlin/models/Payment.kt [hunk 4]
```diff
@@ -7,4 +7,12 @@ val paymentStorage = mutableListOf<Payment>()
```
Comment (popoaq) [inline server/src/main/kotlin/models/Payment.kt:7]: @jw122 one nit, should we move `/models` and `routes` into `com/redenvelope/flagship/`? It seems a bit unusual that they're not under the company domain

**Pull request merged**